lib/chef/provider/aws_launch_configuration.rb

```diff
@@ -10,22 +10,20 @@ def create_aws_object
     image_id = Chef::Resource::AwsImage.get_aws_object_id(new_resource.image, resource: new_resource)
     instance_type = new_resource.instance_type || new_resource.driver.default_instance_type
     options = AWSResource.lookup_options(new_resource.options || options, resource: new_resource)
+    options[:launch_configuration_name] = new_resource.name if !new_resource.name.nil?
+    options[:image_id] = image_id
+    options[:instance_type] = instance_type
 
     converge_by "create launch configuration #{new_resource.name} in #{region}" do
-      new_resource.driver.auto_scaling.launch_configurations.create(
-        new_resource.name,
-        image_id,
-        instance_type,
-        options
-      )
+      new_resource.driver.auto_scaling_client.create_launch_configuration(options)
     end
   end
 
   def update_aws_object(launch_configuration)
     if new_resource.image
       image_id = Chef::Resource::AwsImage.get_aws_object_id(new_resource.image, resource: new_resource)
       if image_id != launch_configuration.image_id
-        raise "#{new_resource.to_s}.image = #{new_resource.image} (#{image.id}), but actual launch configuration has image set to #{launch_configuration.image_id}.  Cannot be modified!"
+        raise "#{new_resource.to_s}.image = #{new_resource.image}, but actual launch configuration has image set to #{launch_configuration.image_id}.  Cannot be modified!"
```

Comment:
`#{new_resource.to_s}`can be simply written as `#{new_resource}` per style

Reply:
@dheerajd-msys: ` new_resource` give the whole details prersent in it but {code} (byebug) new_resource.to_s
"aws_launch_configuration[test_launch_config]"{code} which is required to display resource_type & resource_name